From 69cf9d603426562dba906b72fa959d25c7cedb9e Mon Sep 17 00:00:00 2001
From: Juergen Kussmann <juergen.kussmann@aoe.com>
Date: Wed, 18 Oct 2017 16:17:43 +0200
Subject: [PATCH] [BUGFIX] fix FE-rendering of gridElements (config of
 gridelements could not be determined by loading pages-ts-config)

---
 typo3/sysext/backend/Classes/Utility/BackendUtility.php | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/typo3/sysext/backend/Classes/Utility/BackendUtility.php b/typo3/sysext/backend/Classes/Utility/BackendUtility.php
index e862fe389a..2114427039 100644
--- a/typo3/sysext/backend/Classes/Utility/BackendUtility.php
+++ b/typo3/sysext/backend/Classes/Utility/BackendUtility.php
@@ -4397,7 +4397,12 @@ class BackendUtility
             $wsid = static::getBackendUserAuthentication()->workspace;
         }
         // Check if workspace is different from zero and record is set:
-        if ($wsid !== 0 && is_array($row)) {
+        // BUGFIX:
+        // $wsid will be null if method 'BackendUtility::getPagesTSconfig' will be called in frontend-context without logged-in
+        // BE-user (e.g. gridelements do this to load pages-TS-config - which is needed to render gridelements in frontend)!
+        // If $wsid is null and we wouldn't check that than we would 'override' the pages-record (from LIVE-workspace) with the first
+        // found pages-record, which has 'pid=-1'...so gridelements could use a wrong pages-TS-config to render gridelements in frontend!
+        if ($wsid !== null && $wsid !== 0 && is_array($row)) {
             // Check if input record is a move-placeholder and if so, find the pointed-to live record:
             $movePldSwap = null;
             $orig_uid = 0;
-- 
2.23.0

